conda/core/solve.py

```diff
@@ -1272,27 +1273,83 @@ def _run_solver(self, state):
 
     def _build_transaction(self, state):
         from mamba import mamba_api as api
-        from mamba.utils import to_txn
+        from mamba.utils import to_package_record_from_subjson
+
+        solver = state["solver"]
+        index = state["index"]
+        installed_pkgs = state["installed_pkgs"]
 
         transaction = api.Transaction(
-            state["solver"],
+            solver,
             api.MultiPackageCache(context.pkgs_dirs),
             PackageCacheData.first_writable().pkgs_dir,
         )
         (names_to_add, names_to_remove), to_link, to_unlink = transaction.to_conda()
         specs_to_add = [MatchSpec(m) for m in names_to_add]
         specs_to_remove = [MatchSpec(m) for m in names_to_remove]
 
-        return to_txn(
-            specs_to_add,
-            specs_to_remove,
+        to_link_records, to_unlink_records = [], []
```

Comment:
Wondering if we want to keep the function structure for easier testing?

Reply:
This is going away in #10894 so no need to work on it.